source/common/upstream/outlier_detection_impl.cc

```diff
@@ -607,6 +659,28 @@ void DetectorImpl::processSuccessRateEjections(
       }
     }
   }
+
+  if (!valid_failure_percentage_hosts.empty() &&
+      valid_failure_percentage_hosts.size() >= failure_percentage_minimum_hosts) {
```

Comment:
Don't we also need to check to see if host_monitors_.size() >= failure_percentage_minimum_hosts?
Basically if success_rate_minimum_hosts or failure_percentage_minimum_hosts is low enough we'll gather data for both, but we only want to act on one.

Ditto for avoiding doing work for success rate if success_rate_minimum_hosts is lower

Reply:
I think this is okay since `valid_failure_percentage_hosts` and `valid_success_rate_hosts` are subsets of `host_monitors_`, so `valid_failure_percentage_hosts.size() >= failure_percentage_minimum_hosts` implies `host_monitors_.size() >= failure_percentage_minimum_hosts`.